lib/screens/v2/candidate_generator/dup/alerts.ex

```diff
@@ -127,14 +142,195 @@ defmodule Screens.V2.CandidateGenerator.Dup.Alerts do
+           serialize_map: %{
+             text: %Screens.Config.V2.FreeTextLine{
+               icon: :warning,
+               text: get_kenmore_special_text(branches, :takeover_alert)
+             },
+             header: %{color: :green, text: "Kenmore"},
+             remedy: %Screens.Config.V2.FreeTextLine{
+               icon: :shuttle,
+               text: [%{format: :bold, text: "Use shuttle bus"}]
+             }
+           },
+           widget_type: :takeover_alert,
+           slot_names: [:full_rotation_one]
+         },
+         %DupSpecialCaseAlert{
+           alert_ids: alert_ids,
+           serialize_map: %{
+             text: %Screens.Config.V2.FreeTextLine{
+               icon: :warning,
+               text: get_kenmore_special_text(branches, :partial_alert)
+             },
+             color: :green
+           },
+           widget_type: :partial_alert,
+           slot_names: [:bottom_pane_two]
+         }
+       ]}
+    else
+      {:normal, alerts}
+    end
+  end
+
+  # In the case where we're at WTC and there's a detour affecting routes SL1/2/3, make custom text
+  # Ignore all other SL alerts
+  @spec wtc_special_case(list(Alert.t())) :: {:special, list(Screens.V2.WidgetInstance.t())}
+  def wtc_special_case(alerts) do
+    alert = choose_alert(alerts)
+
+    detoured_routes =
+      alert
+      |> Map.get(:informed_entities)
+      |> Enum.map(fn entity -> if entity.stop === "place-wtcst", do: entity.route end)
+      |> Enum.filter(& &1)
+      |> Enum.sort()
+
+    if detoured_routes === ["741", "742", "743", "746"] do
+      {:special,
+       for rotation_index <- [:full_rotation_zero, :full_rotation_one, :full_rotation_two] do
+         %DupSpecialCaseAlert{
+           alert_ids: [alert.id],
+           serialize_map: %{
+             text: %Screens.Config.V2.FreeTextLine{
+               icon: :warning,
+               text: ["Building closed"]
+             },
+             header: %{color: :silver, text: "World Trade Ctr"},
+             remedy: %Screens.Config.V2.FreeTextLine{
+               icon: :shuttle,
+               text: [%{format: :bold, text: "Board Silver Line on street"}]
+             }
+           },
+           widget_type: :takeover_alert,
+           slot_names: [rotation_index]
+         }
+       end}
+    else
+      {:normal, []}
+    end
+  end
 
-    {:normal, alerts}
+  @spec check_entities_for_stops(Alert.t(), list(%{branch: String.t(), stop: String.t()})) ::
+          atom()
+  def check_entities_for_stops(%{informed_entities: informed_entities}, stop_matchers) do
+    Enum.find(stop_matchers, fn stop ->
+      Enum.any?(informed_entities, fn e ->
+        stop.stop === e.stop
+      end)
+    end)
+    |> Map.get(:branch)
   end
 
-  for {effect, key} <- Enum.with_index([:shuttle, :suspension, :station_closure, :delay]) do
+  @spec get_kenmore_special_text(list(atom()), atom()) :: list(FreeText.t())
+  def get_kenmore_special_text(["b", "c"], :partial_alert),
```

Comment:
Similar to my comment on `serialize_map`, I think the better home for these functions is in the `WidgetInstance`.